iceoryx_utils/include/iceoryx_utils/internal/concurrent/fifo.inl

```diff
@@ -27,11 +27,14 @@ inline bool FiFo<ValueType, Capacity>::push(const ValueType& f_param_r)
     }
     else
     {
-        m_data[m_write_pos.load(std::memory_order_relaxed) % Capacity] = f_param_r;
+        auto currentWritePos = m_write_pos.load(std::memory_order_relaxed);
+        m_data[currentWritePos % Capacity] = f_param_r;
```

Comment:
should we maybe create a local variable named "writeIndex"?

Reply:
I would stick to ```currentWritePos``` since it corresponds to ```m_write_pos```.

Reply:
no, keep both
```
auto currentWritePos = m_write_pos.load(std::memory_order_relaxed);
auto writeIndex = currentWritePos % Capacity;
m_data[writeIndex] = f_param_r;
```